server/src/main/java/org/elasticsearch/gateway/PersistedClusterStateService.java

```diff
@@ -984,6 +1004,24 @@ void commit(long currentTerm, long lastAcceptedVersion, Version oldestIndexVersi
             ensureOpen();
             prepareCommit(currentTerm, lastAcceptedVersion, oldestIndexVersion);
             completeCommit();
+            assert assertOnCommit();
+        }
+
+        private boolean assertOnCommit() {
+            if (assertOnCommit != null /* TODO && Randomness.get().nextInt(100) == 0 */) {
+                // only rarely run this assertion since reloading the whole state can be quite expensive
+                try {
```

Comment:
I guess we can represent the check with one try block

````java
for (var metadataIndexWriter : metadataIndexWriters) {
                    try (var directoryReader = DirectoryReader.open(metadataIndexWriter.indexWriter)) {
                        assertOnCommit.accept(metadataIndexWriter.path, directoryReader);
                    } catch (Exception e) {
                        throw new AssertionError(e);
                    }
}
````

Reply:
Good point, done in 5483687d93c.